<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="scala_takehome.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Take-Home
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

**Import the required libraries**

In [1]:
# import pandas
import pandas as pd

# import numpy
import numpy as np

### Let's begin with some hands-on practice exercises

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. The experiment shows that the average monthly electricity consumption for a house is 458 units. The study records a total of 3206 units of electricity consumption. Calculate the duration of the experiment. Also, identify the criteria to calculate the median consumption. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [2]:
# set the average electricity consumption in 'x_bar'
x_bar = 458

# set the total electricity consumption in 'sum_x'
sum_x = 3206

# the average/mean is calculated as the ratio of total consumption to the number of recorded months (n)
# thus, 'n' can be calculated as the ration of 'sum_x' to 'x_bar'
n = sum_x / x_bar

# print the number of months
print('Duration of the experiment:', n, 'months')

Duration of the experiment: 7.0 months


As n = 7 (odd number), the median consumption is the middlemost observation when the data is arranged in the ascending or descending order.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b> 2. Find the wage above which 65% of wages would lie.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the wages (in dollars) of 15 workers as given below:

            wage = [247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276]

In [3]:
# consider the series of wages
wage = pd.Series([247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276])

# compute 35th percentile using 'quantile'
# As 35th percentile divides the data in 35-65%
wage.quantile(0.35)

284.09999999999997

65% of wages are above 284.1 dollars.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Jimmy requires 38 minutes on average, with a standard deviation of 5 minutes to distribute the newspapers. On the other hand, Richard requires 27 minutes on average, with a standard deviation of 6 minutes to distribute the newspapers. Identify most consistent distributor.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# average time for Jimmy 
jimmy_avg = 38

# standard deviation for Jimmy
jimmy_std = 5

# average time for Richard 
richard_avg = 27

# standard deviation for Richard
richard_std = 6

# calculate coefficient of variation for both the distributors
jimmy_cv = (jimmy_std/jimmy_avg)*100
richard_cv = (richard_std/richard_avg)*100

# print the cv 
print('C.V. for Jimmy:', jimmy_cv)
print('C.V. for Richard:', richard_cv)

C.V. for Jimmy: 13.157894736842104
C.V. for Richard: 22.22222222222222


The output shows that the variation for Jimmy is less than Richard. Thus, Jimmy is a consistent distributor among them.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Identify the extreme weights from the given list of weights using the interquartile range. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the weights (in kg) of 10 students as given below:

            weight = [14.7, 15.2, 39.8, 44.5, 21.2, 28.5, 35.2, 19.7, 40.4, 16.5]

We consider an observation to be an outlier if it is outside the interval (Q1 - 1.5(IQR), Q3 + 1.5(IQR)).

In [5]:
# consider a series of given weights
weight = pd.Series([44.7, 55.2, 49.8, 44.5, 15.2, 78.5, 75.2, 19.7, 50.4, 46.5])

# calculate the first quartile (Q1)
Q1 = weight.quantile(0.25)

# calculate the third quartile (Q3)
Q3 = weight.quantile(0.75)

# calculate the interquartile range (IQR)
IQR = Q3 - Q1

# retrieve the outliers
# pass the condition to detect the outliers
# '|' is used as 'OR' operator on multiple conditions   
outliers = weight[(weight < (Q1 - 1.5 * IQR)) | (weight > (Q3 + 1.5 * IQR))]

# print the weights
outliers

4    15.2
5    78.5
6    75.2
7    19.7
dtype: float64

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. A car showroom in Ohio exhibits a total of 12 cars of different brands. Identify the type of skewness in the showroom prices of cars. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the prices (in dollars) for 12 cars as given below:

            price = [14751, 16422, 15398, 9445, 12589, 11687, 10692, 8475, 11184, 9961, 12898, 11905]

In [6]:
# consider the series of car prices
price = pd.Series([14751, 16420, 15398, 9445, 12589, 11687, 10692, 8775, 11184, 9961, 12898, 11905])

# Karl Pearson's coefficient of skewness is given as,
# skewness = 3(mean - median)/ standard_deviation

# calculate mean
mean = np.mean(price)

# calculate median
median = np.median(price)

# calculate standard deviation
std = np.std(price)

# calculate the coefficient of skewness
skewness = (3.0*(mean - median))/std

# print the value of skewness
print('Coefficient of Skewness:', skewness)

Coefficient of Skewness: 0.45265633664561267


As the coefficient of skewness is positive; the distribution of prices is positively skewed.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. A company offers its customers to take a vehicle and an eduacational loan. A study shows that the probability that a customer takes a vehicle loan is 0.35, the probability that a customer takes an educational loan is 0.46 and the probability that a customer takes at least one of the loan is 0.59. Find the probability that a customer takes both vehicle and educational loans.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [7]:
# consider an event A: Customer takes a vehicle loan
# probability that a customer takes a vehicle loan
prob_vehicle = 0.35

# consider an event B: Customer takes an educational loan
# probability that a customer takes an educational loan
prob_edu = 0.46

# probability that a customer takes one of the loans (P(A ⋃ B))
prob_vehicle_or_edu = 0.59

# to find: probability that a customer takes both the loans (P(A ⋂ B))
# for non mutually exclusive events, P(A ⋃ B) = P(A) + P(B) - P(A ⋂ B)
# P(A ⋂ B) = P(A) + P(B) - P(A ⋃ B) 
prob_vehicle_and_edu = prob_vehicle + prob_edu - prob_vehicle_or_edu

print('The probability that a customer takes both the loans:', prob_vehicle_and_edu)

The probability that a customer takes both the loans: 0.22000000000000008


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Consider two independent events X and Y. If the probability of occurrence of event X is 0.4 and the probability of occurrence of at least one event is 0.6. Find the probability of occurrence of event Y. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [8]:
# probability of occurrence of event X
prob_X = 0.4

# probability of occurrence of at least one event
prob_X_or_Y = 0.6

# Since, X and Y are independent events P(X ⋂ Y) = P(X).P(Y)
# i.e. P(X ⋃ Y) = P(X) + P(Y) - P(X).P(Y)
# P(Y) = (P(X ⋃ Y) - P(X)) / (1 - P(X))
prob_Y = (prob_X_or_Y - prob_X) / (1 - prob_X)

print('Thr probability of occurrence of event Y:', prob_Y)

Thr probability of occurrence of event Y: 0.33333333333333326


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Consider a vending machine containing 6 boxes each of four different juice flavors (Mango, Apple, Guava and Kiwi). Find the probability that the juice box picked is either of Mango or Guava flavor.                 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [9]:
# number of boxes of each flavor is 6
boxes = 6

# total number of boxes in a vending machine
total_boxes = 6*4

# probability that the box picked is of mango or guava juice
# req_prob = (probability that the box is of mango juice) + (probability that the box is of guava juice) 
prob_mango = boxes / total_boxes
prob_guava = boxes / total_boxes

req_prob = prob_mango + prob_guava

print('The probability that the juice box picked is either of Mango or Guava flavor:', req_prob)

The probability that the juice box picked is either of Mango or Guava flavor: 0.5


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Consider a student appearing for 3 different exams either pass or fail in each exam. Let a marginal probability of a student passing the Economics, Mathematics and German exam be 0.67. The probability that the student fails in the Economics exam is 0.10 and the probability that the student fails in the German exam is 0.09. Find the probability that a student fails in the Mathematics exam.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Consider a discrete variable `X` denoting the passing or failing of an exam and a discrete variable `Y` denoting the three subjects of the exam. The sum of marginal probabilities of X (or Y) should be always equal to 1.

In [10]:
# let 'X_pass' and 'X_fail' be the unique values in the variable X
# the marginal probability that a student passes all the three exams 
prob_X_pass = 0.67

# as, the sum of marginal probabilities for 'prob_X_pass' and 'prob_X_fail' should be 1
prob_X_fail = 1 - prob_X_pass

# the marginal probability 'prob_X_fail' is the sum of probabilities that a student fails in all the three exams
# let 'prob_eco_fail', 'prob_maths_fail' and 'prob_german_fail' be the probabilities that a student fails in the exam
# from given data
prob_eco_fail = 0.10
prob_german_fail = 0.09

# to find: prob_maths_fail
# prob_X_fail = prob_eco_fail + prob_maths_fail + prob_german_fail
prob_maths_fail = prob_X_fail - prob_eco_fail - prob_german_fail 

print('The probability that a student fails in Mathematics exam', prob_maths_fail)

The probability that a student fails in Mathematics exam 0.13999999999999996


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. The probabilities that a machine will produce 2, 5, 7, 9 defective batteries in a day are 0.3, 0.22, 0.15, 0.1 respectively. Find the expected number of defective batteries produced in a day.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [11]:
# expected number of defective batteries produced is the sum of product of number of batteries and the respective probability
X = np.array([2, 5, 7, 9])
prob_X = np.array([0.3, 0.22, 0.15, 0.1])

# caclulate the expected number of defective batteries
expected_value = np.sum(X*prob_X)

print('Expected number of defective batteries produced in a day:', expected_value)

Expected number of defective batteries produced in a day: 3.65


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. In a pack of 52 cards, what is the probability of selecting a red card?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [12]:
# from a pack of 52 cards, the chance of selecting any card is equally likely
probability_of_selecting_card = 1/52

# a pack of cards has 26 red cards
required_probability = 26 * probability_of_selecting_card

print('The probability of selecting a red card', required_probability)

The probability of selecting a red card 0.5


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. From the experience, it is seen that 4% of the tyres produced by the machine are defective. Out of the 10 tyres produced, find the probability that at most 3 are defective.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [13]:
# import the required library
from scipy.stats import binom

# X: the number of defective tyres produced 
# n = 10
# p = 0.04
# required probability is P(X ≤ 3) = P(X = 0) + P(X = 1) + P(X = 2) + P(X = 3)
req_prob = binom.pmf(0, 10, 0.04) + binom.pmf(1, 10, 0.04) + binom.pmf(2, 10, 0.04) + binom.pmf(3, 10, 0.04)

print('The probability that at most 3 tyres out of 10 are defective:', req_prob)

The probability that at most 3 tyres out of 10 are defective: 0.9995574006400002


In [14]:
# or one may use the cdf() and pass the k to get P(X ≤ k)
binom.cdf(3, 10, 0.04) 

0.9995574006399999

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. The number of trucks crossing a bridge during the day follow a Poisson distribution with mean 18. What is the probability that on a randomly selected day 12 trucks would have crossed the bridge?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [15]:
# import the required library
from scipy.stats import poisson

# X: number of trucks crossing a bridge during the day
# x = 12
# m = 18
# The required probability is P(X = 12)
req_prob = poisson.pmf(12, 18)

print('The probability that on a randomly selected day 12 trucks would have crossed the bridge:', req_prob)

The probability that on a randomly selected day 12 trucks would have crossed the bridge: 0.03678175293604137


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. A random variable follows uniform distribution in the interval 5 to 10. Obtain the mean and variance. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [16]:
# lower limit of the uniform distribution 
lower_limit = 5

# upper limit of the uniform distribution 
upper_limit = 10

In [17]:
# compute the mean of the uniform distribution
uniform_mean = (upper_limit - lower_limit) / 2

# print the mean
print('Mean:', uniform_mean)

Mean: 2.5


In [18]:
# compute the variance of the uniform distribution
uniform_variance = (upper_limit - lower_limit)**2 / 12

# print the variance
print('Variance', uniform_variance)

Variance 2.0833333333333335


<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Consider an random variable that follows normal distribution with mean 5 and variance 10 
obtain: <br>
1. P(X < 1)<br>
2. P(X > 5)<br>
3. P(3 < X < 7)
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [19]:
# import the required library
from scipy.stats import norm

# mean of the distribution 
mu = 5 

# variance of the distribution 
var =10

# standard deviation of the distribution 
sd = np.sqrt(var)

In [20]:
# 1. P(X < 1)
# use cdf() to calculate P(X < k)
norm.cdf(1, mu, sd)

0.1029516053660342

In [21]:
# 2. P(X > 5)
# use cdf() to calculate P(X < k)
1 - norm.cdf(5, mu, sd)

0.5

In [22]:
# 3. P(3 < X < 7)
# use cdf() to calculate P(X < k)
norm.cdf(6, 3, 2) - norm.cdf(4, 3, 2)

0.2417303374571288

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Consider the data given below. Can we claim that the average number of fruits on a tree is 70? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:

            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

Let <br>X: Number of fruits on a tree <br>
$\mu$: Average number of fruits on a tree  <br>

The hypothesis is                                                                       
H<sub>0</sub>: The average number of fruits on a tree is equal to 70, i.e $\mu$= 70 <br>
**AGAINST**<br>
H<sub>a</sub>: The average number of fruits on a tree is not equal to 70, , i.e $\mu$ ≠ 70

In [23]:
# create the data series
number_fruits = pd.Series([64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46])

# t-test for one sample
from scipy.stats import ttest_1samp

# conduct the test
ttest_1samp(number_fruits, 70)

Ttest_1sampResult(statistic=-1.36854291090804, pvalue=0.19270344894968694)

We see that the p-value (0.1927) is greater than 0.05 (level of significance), hence we accept H<sub>0</sub>.<br> We may conclude that the average number of fruits on a tree is 70.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. The courier company has two offices in NYC. They are interested to know which office is busier. The number of parcels received for 15 days is given below. Can we claim that both offices are equally busy?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the number of parcels in office A:
            office_A = [26, 56, 48, 56, 79, 67, 35, 46, 87, 53, 67, 89, 65, 42, 34]

            Consider the number of parcels in office B:
            office_B = [34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95]

Let <br>X: Number of parcels in office A on a particular day <br>
    Y: Number of parcels in office B on a particular day <br><br>
Thus <br>
$\mu$<sub>A</sub>: Average number of parcels in office A<br>
$\mu$<sub>B</sub>: Average number of parcels in office B <br>
    
                                                                  
The hypothesis is                                                                     
H<sub>0</sub>: The average number of parcels in office A is equal to the average number of parcels in office B, i.e $\mu$<sub>A</sub> = $\mu$<sub>B</sub> <br>
**AGAINST**<br>
H<sub>a</sub>: The average number of parcels in office A is not equal to the average number of parcels in office B, i.e $\mu$<sub>A</sub> ≠ $\mu$<sub>B</sub> <br>

In [24]:
# create the data series
# parcels in office A
parcels_office_A = pd.Series([64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46])

# parcels in office B
parcels_office_B = pd.Series([34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95])

# t-test for testing equality of means
from scipy.stats import ttest_ind

# conduct the test
ttest_ind(parcels_office_A, parcels_office_B)

Ttest_indResult(statistic=0.6449141708280643, pvalue=0.5242321636740674)

We see that the p-value (0.524) is more than 0.05 (level of significance), hence we accept H<sub>0</sub>.<br> We may conclude that the average number of parcels in office A is equal to the average number of parcels in office B.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. To improve the attitude of officers, they had to complete a training program. Can we claim that the training program was effective? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

The data of 10 officers scores is given below:

<table>
    <tr>
        <th> Officer </th>
        <th> Attitude Before </th>
        <th> Attitude After </th>
    </tr>
     <tr>
        <td> 1 </td>
        <td> 3 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 2 </td>
        <td> 5 </td>
        <td> 5 </td>
    </tr>
    <tr>
        <td> 3 </td>
        <td> 2 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 4 </td>
        <td> 2 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 5 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 6 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 7 </td>
        <td> 3 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 8 </td>
        <td> 5 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 9 </td>
        <td> 3 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 10 </td>
        <td> 4 </td>
        <td> 5 </td>
    </tr>
</table>

Let <br>X: Score before training <br>
    Y: Score after training<br><br>
<br>
$\mu$<sub>X</sub>: Average score before training<br>
$\mu$<sub>Y</sub>: Average score after training <br>
    
                                                                  
The hypothesis is                                                                     
H<sub>0</sub>: The average score before training is equal to the average score after training, i.e $\mu$<sub>X</sub> = $\mu$<sub>Y</sub> <br>
**AGAINST**<br>
H<sub>a</sub>:  The average score before training is not equal to the average score after training, i.e $\mu$<sub>X</sub> ≠ $\mu$<sub>Y</sub> 

In [25]:
# create the data
# score before training
score_before = pd.Series([3, 5, 2, 2, 4, 4, 3, 5, 3, 4])

# score after training
score_after = pd.Series([4, 5, 3, 4, 4, 4, 3, 2, 2, 5])

# import the 'ttest_rel' from library Scipy
from scipy.stats import ttest_rel

# conduct the paired t-test
ttest_rel(score_before, score_after)

Ttest_relResult(statistic=-0.2307692307692308, pvalue=0.8226545408271233)

We see that the p-value (0.822) is more than 0.05 (level of significance), hence we accept H<sub>0</sub>.<br> We may conclude that there is no significant effect of the training.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. During an economic crisis, a researcher thought that the salaries of IT professionals are affected. Test whether the average monthly salary has gone below 65000 dollars.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Salaries of corporate professionals in thousand dollars:

            salary = [56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 
                      42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65,
                      56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65]

Let <br>X: Salary of a corporate professional <br>
$\mu$: Average salary of a corporate professional  <br>

The hypothesis is                                                                       
H<sub>0</sub>: The average salary of a corporate professional is equal to 65000, i.e $\mu$= 65000 <br>
**AGAINST**<br>
H<sub>a</sub>: The average salary of a corporate professional is less than 65000, , i.e $\mu$ ≤ 65000

In [26]:
# create the data series
salaries = pd.Series([56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 
                      42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65,
                      56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65])


# import the library
from statsmodels.stats import weightstats as stests

# conduct the test
# since the data is scaled, we shall consider 65 instead of 65000
# only for ztest() in the library statsmodels we can set the hyperparameter 'alternative'
stests.ztest(salaries, value = 65, alternative = "smaller")

(-1.1250934292594312, 0.1302747226990263)

We see that the p-value (0.13) is more than 0.05 (level of significance), hence we accept H<sub>0</sub>.<br> We may conclude that there is no change in the salaries of the professionals.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Consider the data given below. Can we conclude that the data is normally distributed?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:
            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

The null and alternate hypothesis of Shapiro-Wilk test are as follows: <br>
    
H<sub>0</sub>: The data is normally distributed<br>
**AGAINST**<br>
H<sub>a</sub>: The data is not normally distributed

In [27]:
# create the data series
number_fruits = pd.Series([64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46])

# import shapiro from the scipy library
from scipy.stats import shapiro

# conduct the normality test
stat, p = shapiro(number_fruits)

print('The test statistic is', stat, 'and its p-value is', p)

The test statistic is 0.9354038238525391 and its p-value is 0.3280726969242096


Since the p-value (0.328) is more than 0.05 (level of significance), hence we accept H<sub>0</sub>. <br>We may conclude that the data is normally distributed.